### PR TITLE
Clean up the usage of kuler90/build-unity

### PR DESCRIPTION
**Pull request by sunmou99**
### Description
`u3d` is broken. 
@a-maurice found out a workaround (replacing `u3d` with `kuler90/build-unity`): https://github.com/firebase/firebase-unity-sdk/pull/532
This is a cleaning up version of @a-maurice's workaround. 

TODO: We are still depend on another tool for unity installation. Need to have more control & design for Unity Installation later.
***
### Testing
Packaging SDK: https://github.com/firebase/firebase-unity-sdk/actions/runs/3466176603
Integration Test: https://github.com/firebase/firebase-unity-sdk/actions/runs/3466177578

[replace this line]: # (Describe your testing in detail.)
***

### Type of Change
Place an `x` the applicable box:
- [ ] Bug fix. Add the issue # below if applicable.
- [ ] New feature. A non-breaking change which adds functionality.
- [x] Other, such as a build process or documentation change.
***

### CODE REVIEW

File: .github/workflows/integration_tests.yml [hunk 1]
```diff
@@ -218,7 +218,7 @@ jobs:
           fi
           python scripts/gha/build_testapps.py \
             --t ${{ needs.check_and_prepare.outputs.apis }} \
-            --u "${{ steps.unity_setup.outputs.unity_version }}" \
+            --u ${{ env.UNITY_VERSION }} \
```
Comment (a-maurice) [inline .github/workflows/integration_tests.yml:221]: Where is this getting set?

File: gha/unity/unity_installer.py [hunk 2]
```diff
@@ -313,26 +279,38 @@ def get_unity_executable(version):
   # https://github.com/DragonBox/u3d
```
Comment (a-maurice) [inline gha/unity/unity_installer.py:279]: Remove old comment

File: gha/unity/action.yml [hunk 3]
```diff
@@ -46,33 +42,26 @@ runs:
...
-    - name: Install Unity installer (U3D)
-      if: inputs.release_license == ''
-      uses: nick-invision/retry@v2
-      with:
-        timeout_minutes: 10
-        max_attempts: 2
-        shell: bash
-        command: gem install u3d -v 1.2.3
-    - name: Set the Unity Env
+    - name: unity_setting
```
Comment (a-maurice) [inline gha/unity/action.yml:45]: Doesn't need to be done now, but after spending some time with this action, I think this should be split into 3 separate ones.
1. Install
2. Activate
3. Deactivate

Doing it based on input parameters is a bit confusing, especially at a glance.

File: .github/workflows/integration_tests.yml (hunk 1 above)
Comment (sunmou99) [reply .github/workflows/integration_tests.yml:221]: https://github.com/firebase/firebase-unity-sdk/blob/88d88cba279144e214ea8a4d894250d8d8c52306/gha/unity/action.yml#L50

File: gha/unity/unity_installer.py (hunk 2 above)
Comment (sunmou99) [reply gha/unity/unity_installer.py:279]: Will do.

File: gha/unity/action.yml (hunk 3 above)
Comment (sunmou99) [reply gha/unity/action.yml:45]: I just want to make the usage simpler: https://github.com/firebase/firebase-unity-sdk/tree/main/gha/unity#usage
But I will add more variables in this action and make the action more descriptive.

File: .github/workflows/integration_tests.yml [hunk 4]
```diff
@@ -422,23 +358,14 @@ jobs:
...
           pip install -r scripts/gha/requirements.txt
-      - name: Setup Unity
-        uses: kuler90/setup-unity@v1
+      - id: unity_setup
+        uses: ./gha/unity
         timeout-minutes: 30
         with:
-          unity-version: 2020.3.34f1
-          # Install ios for the editor tools
-          unity-modules: ios
```
Comment (a-maurice) [inline .github/workflows/integration_tests.yml:431]: For playmode tests I had to install the iOS tools for Crashlytics and Messaging. Can you add this back in, and make sure that those tests pass.

Comment (sunmou99) [reply .github/workflows/integration_tests.yml:431]: Is the iOS tool required only on Windows machine?

Comment (sunmou99) [reply .github/workflows/integration_tests.yml:431]: It seems that the ios module only required on Windows machines.
Test result: https://github.com/firebase/firebase-unity-sdk/actions/runs/3473188998

**Review by a-maurice — APPROVED** [review]

**Pull request merged**